### PR TITLE
New Architecture landing page

### PR DESCRIPTION
**Pull request by lindboe**
Adds content to the New Architecture landing page, which should introduce the New Architecture and direct the reader to the relevant content for them.

### CODE REVIEW

File: docs/the-new-architecture/landing-page.md [hunk 1]
```diff
@@ -3,11 +3,26 @@ id: landing-page
...
+- improved type safety and consistency of communication between JavaScript and native code
 
-- New user: link to how to use the new [app template](use-app-template)
-- New Library developer: link to the [pillars](pillars)
-- Old library developer/app developer: link to [the migration guide](../new-architecture-intro)
+This is made possible by these [Pillars of the New Architecture](pillars):
+
+- [Fabric renderer and components](pillars-fabric-components), and the capabilities they offer for layout and interaction events
+- [TurboModules](pillars-turbo-modules), which supports faster loading of and communication with native code
+- [CodeGen](pillars-codegen), which generates code you can use to interface with native modules, via static typing in JavaScript
```
Comment (lindboe) [inline docs/the-new-architecture/landing-page.md:16]: I'm not very confident in the story we want to tell here. Let me know if there's a better way to frame the pitch for any of this.

I pulled the benefits largely from https://reactnative.dev/architecture/fabric-renderer.

As each named piece we wanted to mention, I did my best to quickly summarize what they provide, but I'm not totally sure about a few things:
* I've mostly read about Fabric as a _renderer_ and the idea of "Fabric components" is new to me. Does that just mean any component that uses a Fabric capability, or any component that's compatible with Fabric?
* I understand the idea and benefits behind JSI, but I'm not sure I'm correctly summarizing what CodeGen does.
* Similarly with TurboModules, I don't understand well yet everything the concept contains. I based this summary off the understanding I got from previously reading https://formidable.com/blog/2019/fabric-turbomodules-part-3/.

File: docs/the-new-architecture/landing-page.md [hunk 2]
```diff
@@ -3,11 +3,26 @@ id: landing-page
 title: Introduction
 ---
 
-This section is the entry point for the new guide’s documentation.
-It contains some basic information about the New Architecture: its pillars, the version from which it has been made available and other very high-level information. Then, it should present the guide’s structure itself.
+Starting in version 0.68, React Native provides a New Architecture that improves several aspects of how JavaScript code and host platforms can integrate, such as:
 
-This section should also contains a sort of Driving guide based on the user use-cases:
+- synchronously measure and render React Native views at the same time as native views
```
Comment (lindboe) [inline docs/the-new-architecture/landing-page.md:8]: I noticed that the Architecture section on Fabric uses "host platform" and "host views", but I'm a little unsure when it's more appropriate to refer to "native code" instead (something I noticed STYLEGUIDE.md also discourages when possible).

Comment (cipolleschi) [reply docs/the-new-architecture/landing-page.md:8]: @cortinico any suggestion here?

File: docs/the-new-architecture/landing-page.md [hunk 3]
```diff
@@ -3,11 +3,26 @@ id: landing-page
...
+- support for prioritization of events that update the app's layout, like user interactions
+- improved type safety and consistency of communication between JavaScript and native code
```
Comment (cipolleschi) [inline docs/the-new-architecture/landing-page.md:10]: You can pick some other benefits from [here](https://github.com/facebook/react-native-website/pull/3043). It also contains a slightly more detailed explanation about why we are migrating to the new architecture.

File: docs/the-new-architecture/landing-page.md (hunk 1 above)
Comment (cipolleschi) [reply docs/the-new-architecture/landing-page.md:16]: [Here](https://github.com/facebook/react-native-website/pull/3046) I wrote about the main pillars.
Fabric is the name of the Renderer. A Fabric Component (FC) is a UI component that leverage the Fabric Renderer.
TurboModules (TM) are libraries that don't have UI parts and that leverages the JSI mechanism to communicate between JavaScript and the platform code.

CodeGen is a meta-pillar: we introduced that section in the Pillars because it is important that we explain what it does and ow it works after the reader has some concepts about the TM and FC development cycle. It is not a real pillar because, ideally, it must stay hidden from the users' eyes. However, we need to document it because they have to write code keeping the generated code into consideration.

I hope that this short explanation helps you with these concepts. Feel free to ask more questions or we can have a meeting if that could speed you up.

File: docs/the-new-architecture/landing-page.md [hunk 4]
```diff
@@ -3,11 +3,26 @@ id: landing-page
...
+
+### For app developers
+
+To **create a new app** using the New Architecture, head over to [Creating a New Architecture App](use-app-template.md), which will get you up and running in a few easy steps with the new app template.
+
+To **migrate an existing app** to the New Architecture, follow [Adopting the New Architecture](../new-architecture-intro.md).
+
+### For library maintainers
+
+If you are adding support for the New Architecture to a library, head to the [Pillars]([pillars]) section to learn more about supporting apps that have migrated to the New Architecture, while still maintaining backwards compatibility for apps that haven't.
```
Comment (cipolleschi) [inline docs/the-new-architecture/landing-page.md:28]: Backward compatibility will have a separate section.
I think that here we are missing two paths:
* Do you already have a library and you want to migrate to the new architecture => go to the library migration guide
* Do you already have a library and it has to work on both the architectures => go to the backward compatibility guide.

We kept the backward compatibility separated with respect to the creation of a TM or FC because it requires some extra steps. Going forward, we would like to have all the users on the New Architecture and to have only TM and FC that works with the New Architecture.

**Review by cipolleschi — CHANGES_REQUESTED** [review]
Splendid work summarizing the concepts. I left you more references to help you: they could spark ideas on how to improve the current document.
I think we are missing a couple of pointers in the last part of the document.

File: docs/the-new-architecture/landing-page.md (hunk 1 above)
Comment (lindboe) [reply docs/the-new-architecture/landing-page.md:16]: > A Fabric Component (FC) is a UI component that leverage the Fabric Renderer.

What is meant by "leverage" is still ambiguous to me: does that mean that the component is:
a) simply compatible with the Fabric renderer (e.g., my custom component that just renders a customized `<View>` would be a Fabric component, since `View` is compatible with Fabric), or 
b) that it implements code that directly interfaces with native views (e.g., a "host component" as defined in https://reactnative.dev/architecture/glossary#react-host-components-or-host-components), or 
c) that it specifically utilizes capabilities that are unique to Fabric?

> TurboModules (TM) are libraries that don't have UI parts

That is an interesting distinction, I don't think I understood that before. Why do they specifically not have UI parts? Because that would be considered a Fabric component?

On CodeGen, I think I feel less confident in describing it because I don't know the concrete details of what is actually being generated and how that generated code is used. I think in part I need to try it out.

File: docs/the-new-architecture/landing-page.md (hunk 3 above)
Comment (lindboe) [reply docs/the-new-architecture/landing-page.md:10]: The explanation in that document is thorough, but concise. Given that it's not very long, I'm wondering how much is worth repeating vs. just expressing something along the lines of "The New Architecture provides developers more tools for creating performant and responsive apps. Check out 'Why a New Architecture' for more information."

**Comment by lindboe** [comment]
Pushed a few updates, but I stopped on updating the brief descriptions of the pillars for now, since I asked a few more questions; I'm interested in what y'all think are the most important things to say about each in a one-line summary

File: docs/the-new-architecture/landing-page.md (hunk 1 above)
Comment (cipolleschi) [reply docs/the-new-architecture/landing-page.md]: > What is meant by "leverage" is still ambiguous to me: does that mean that the component is:

RN with the New Architecture, we have two different renderers: Paper (the old architecture) and Fabric (the new architecture). A renderer is a piece of code that draws elements on the screens.
* Native Components (from the old architecture) use Paper to be drawn on the screen.
* Fabric Components use Fabric. 

I think that this is the easiest way to describe the difference. Technically, to actually use Farbic, a Fabric Component has to implement a C++ interface (whose boilerplate is automatically generated) and it has to be registered in a special component registry (done automatically for iOS, it requires a little bit of manual steps for Android).

> That is an interesting distinction, I don't think I understood that before. Why do they specifically not have UI parts? Because that would be considered a Fabric component?

TL;DR: yes. 
The long answer is that Fabric is tightly bound to the UI and how we draw elements on the screen. However, not all the libraries requires to draw stuff on the screen (imagine a Networking library, for example). In those case, it does not make sense to download and install all the UI related dependencies. To allow this optimization, we split TM and FC in two different pillars. 

> On CodeGen, I think I feel less confident in describing it because I don't know the concrete details of what is actually being generated and how that generated code is used. I think in part I need to try it out.

That make sense. However, for this page, I think that it's enough to specify that the codeGen is an automatic steps that pregenerate a lot of C++ boilerplate code that is required by the new architecture. Ideally, devs should never see the automatically generated code. But it is not like that at the moment, so we need to at least cite it.

File: docs/the-new-architecture/landing-page.md (hunk 3 above)
Comment (cipolleschi) [reply docs/the-new-architecture/landing-page.md]: > "The New Architecture provides developers more tools for creating performant and responsive apps. Check out 'Why a New Architecture' for more information."

Great suggestion. Go for it! :+1:

File: docs/the-new-architecture/landing-page.md [hunk 5]
```diff
@@ -3,11 +3,26 @@ id: landing-page
 title: Introduction
 ---
 
-This section is the entry point for the new guide’s documentation.
-It contains some basic information about the New Architecture: its pillars, the version from which it has been made available and other very high-level information. Then, it should present the guide’s structure itself.
+Starting in version 0.68, React Native provides the New Architecture, which offers developers new capabilities for building highly performant and responsive apps. Visit [Why a New Architecture](why) to learn more about what drove the decision to re-architect, and the benefits it provides.
 
-This section should also contains a sort of Driving guide based on the user use-cases:
+The core concepts are explained by the [Pillars of the New Architecture](pillars):
```
Comment (cipolleschi) [inline docs/the-new-architecture/landing-page.md:8]: I would rephrase this slightly. It looks a little bit abrupt moving from the `why` to the `components` like that.

We could add a sentence on the lines of:
> In order to achieve these benefits, we had to rethink how Native Modules and Native Components works. This led us to develops the [Pillars of the New Architecture](pillars):

What do you think?

Comment (lindboe) [reply docs/the-new-architecture/landing-page.md:8]: Yeah, good idea!

File: docs/the-new-architecture/landing-page.md (hunk 2 above)
Comment (cortinico) [reply docs/the-new-architecture/landing-page.md]: I saw this got removed so my input is probably not needed anymore 👍

**Review by cortinico — APPROVED** [review]
I went through the convos in this PR and I'm aligned on what was mentioned 👍 @cipolleschi already did a deep review and I don't have anything to add, other than thanking you for the critical piece of documentation you added

File: docs/the-new-architecture/landing-page.md (hunk 1 above)
Comment (lindboe) [reply docs/the-new-architecture/landing-page.md]: Alright, I tried rephrasing to what I think adds more relevant but broad context. Let me know whether you think it's an improvement.

**Review by cipolleschi — APPROVED** [review]
